installer/install.iss

```diff
@@ -1215,21 +1215,26 @@ begin
      *)
 
     Path:=EditorPage.Values[0]+' ';
-    if not WildcardMatch(Lowercase(Path), '"*.exe" *') then
-        PathLength:=Pos('.exe ',Lowercase(Path))+3
-    else begin
-        PathLength:=Pos('.exe" ',Lowercase(Path))+2;
-        Path:=Copy(Path,2,PathLength)+Copy(Path,PathLength+3,Length(Path))
-    end;
+    if Pos('.exe',Lowercase(Path))=0 then begin
+    	CustomEditorPath:=EditorPage.Values[0];
+        CustomEditorOptions:='';
```

Comment:
This would neatly fit in the previous code's `else` clause:

```pascal
    if not WildcardMatch(Lowercase(Path), '"*.exe" *') then
        PathLength:=Pos('.exe ',Lowercase(Path))+3
    else if Pos('.exe',Lowercase(Path))=0 then begin
        CustomEditorPath:=EditorPage.Values[0];
        CustomEditorOptions:='';
    else begin
        PathLength:=Pos('.exe" ',Lowercase(Path))+2;
        Path:=Copy(Path,2,PathLength)+Copy(Path,PathLength+3,Length(Path))
    end;
```

But what about `code --wait`? It would fall into that new clause, and assume that the entire string is the path to the executable.

I would think, however, that it would maybe make more sense to replace the `"*.exe" *` pattern by `"*" *`, and then just search for `" ` instead of `.exe ` in line 1221.

But I guess the entire code is confusing, because the first clause is about *not* matching that pattern. So how about this:

```pascal
    if WildcardMatch(Lowercase(Path), '"*" *') then
        PathLength:=Pos('" ',Lowercase(Path));
        CustomEditorPath:=Copy(Path,2,PathLength);
        CustomEditorOptions:=Copy(Path,PathLength+2,Length(Path))
    else if Pos(' ',Lowercase(Path))>0 then begin
        PathLength:=Pos(' ',Lowercase(Path));
        CustomEditorPath:=Copy(Path,1,PathLength);
        CustomEditorOptions:=Copy(Path,PathLength+2,Length(Path))
    else begin
        CustomEditorPath:=Path;
        CustomEditorOptions:=''
    end;
```

Reply:
Currently the text field supports path with spaces (Like - C:\Program Files\editor.exe -w) but in your implementation, it would fall in the second clause making things go wrong.

I can surely make this nested if to if..elseif..else way you suggested. I will make this change in next commit.

Reply:
> Currently the text field supports path with spaces (Like - C:\Program Files\editor.exe -w) but in your implementation, it would fall in the second clause making things go wrong.

That's a good point. So we should first check for quoted `.exe`, then unquoted `.exe`, then quoted path without extension, then unquoted path without extension, right?